src/otx/engine/engine.py

```diff
@@ -804,6 +807,14 @@ def _build_trainer(self, **kwargs) -> None:
         """Instantiate the trainer based on the model parameters."""
         if self._cache.requires_update(**kwargs) or self._trainer is None:
             self._cache.update(**kwargs)
+            # set up xpu device
+            if self._device.accelerator == DeviceType.xpu:
+                self._cache.update(strategy="xpu_single")
+                # add plugin for Automatic Mixed Precision on XPU
+                if kwargs["precision"] == 16:
+                    self._cache.update(plugins=[MixedPrecisionXPUPlugin()])
+                    self._cache.args["precision"] = None
```

Comment:
This probably won't cope with the case where `kwargs["precision"] = None` and `self._cache.args["precision"] = 16`.

Reply:
[Done](https://github.com/openvinotoolkit/training_extensions/pull/3094/commits/7e303a1a9caedfa247c9c98d47e7956d5fee01d0)